src/events/guildMemberUpdate.ts

```diff
@@ -1,34 +1,35 @@
-const Discord = require('discord.js');
-const util = require('../util');
+import { AuditLogEvent, EmbedBuilder } from 'discord.js';
+import { sendEventLogMessage } from '@/util';
+import type { GuildMember, PartialGuildMember } from 'discord.js';
 
-/**
- * 
- * @param {Discord.GuildMember} oldMember
- * @param {Discord.GuildMember} newMember
- */
-async function guildMemberUpdateHandler(oldMember, newMember) {
-	if (oldMember.user.bot) return;
+export async function guildMemberUpdateHandler(oldMember: GuildMember | PartialGuildMember, newMember: GuildMember): Promise<void> {
+	if (oldMember.user.bot) {
+		return;
+	}
 
 	const member = newMember;
 	const guild = member.guild;
 	const user = member.user;
 
 	const auditLogs = await guild.fetchAuditLogs({
 		limit: 1,
-		type: 'MEMBER_UPDATE'
+		type: AuditLogEvent.MemberUpdate
 	});
 
 	const latestLog = auditLogs.entries.first();
+	if (!latestLog) {
+		return;
+	}
 
 	const { executor } = latestLog;
 
-	const eventLogEmbed = new Discord.MessageEmbed();
+	const eventLogEmbed = new EmbedBuilder();
 
 	eventLogEmbed.setColor(0xC0C0C0);
 	eventLogEmbed.setDescription('――――――――――――――――――――――――――――――――――');
 	eventLogEmbed.setFooter({
 		text: 'Pretendo Network',
-		iconURL: guild.iconURL()
+		iconURL: guild.iconURL() as string
```

Comment:
`guild.iconRL()` returns a `string | null` and `iconURL` needs a `string | undefined`. Also, we know this will always exist so this is just the nicest way I could think of to not complain. Happy to change this if there's something you'd prefer

Reply:
I'll update to that

Reply:
So it turns out I'd already done this in most places, I've made so many changes for this PR I'm forgetting them in my old ages 😂